PROTOCOL.md

```diff
@@ -995,6 +1065,178 @@ Within the checkpoint, the `remove` struct does not contain the `stats` and `tag
+
+3. Not amenable to performance and scalability optimizations. For example, there is no way to store skipping stats for checkpoint parts, nor to reuse checkpoint part files across multiple checkpoints.
+
+4. Multi-part checkpoints also bloat the _delta_log dir and slow down LIST.
+
+[UUID-named](#uuid-named-checkpoint) checkpoint (which follow [V2 spec](#v2-spec)) solves all
+of these problems and should be preferred over multi-part checkpoints. For this reason, Multi-part
+checkpoints are forbidden when [V2 Checkpoints table feature](#v2-checkpoint-table-feature) is enabled.
+
+### Handling Backward compatibility while moving to UUID-named v2 Checkpoints
+
+A UUID-named v2 Checkpoint should only be created by clients if the [v2 checkpoint table feature](#v2-checkpoint-table-feature) is enabled.
+When UUID-named v2 checkpoints are enabled, Writers should occasionally create a v2 [Classic Checkpoint](#classic-checkpoint)
+for maintaining compatibility with older clients. These classic checkpoints have the same content as the UUID-named v2 checkpoint
+file name, but older clients (those which do not support [v2 checkpoint table feature](#v2-checkpoint-table-feature)) will recognize the classic
+file name, allowing them to extract [Protocol](#protocol-evolution) and fail gracefully with an invalid protocol
+version error on v2-checkpoint enabled tables. Writers should create classic checkpoints often enough to allow older
+clients to discover them and fail gracefully.
+
+### Allowed combinations for `checkpoint spec` <-> `checkpoint file naming`
+
+Checkpoint Spec | [UUID-named](#uuid-named-checkpoint) | [classic](#classic-checkpoint) | [multi-part](#multi-part-checkpoint)
+-|-|-|-
+[V1](#v1-spec) | Invalid | Valid | Valid
+[V2](#v2-spec) | Valid | Valid | Invalid
+
+### Metadata Cleanup
+
+The _deltaLog directory could increase in size as more and more commits and checkpoints are accumulated.
+Implementations are recommended to do regular cleanup of old deltaLog to cleanup unneeded files and reduce the directory size.
+The following steps could be used to do cleanup of the DeltaLog directory:
+1. Identify a threshold (in days) uptil which we want to preserve the deltaLog. Let's refer to
+midnight UTC of that day as `cutOffTimestamp`. The newest commit not newer than the `cutOffTimestamp` is
+the `cutoffCommit`, because a commit exactly at midnight is an acceptable cutoff. We want to retain everything including and after the `cutoffCommit`.
+2. Identify the most recent checkpoint before the `cutOffCommit`. Lets call it `cutOffCheckpoint`.
+We need to preserve the `cutOffCheckpoint` to enable time travel for commits between `cutOffCheckpoint`
+and the next available checkpoint.
```

Comment:
Two things we could convey more clearly:
1. We want to identify the newest checkpoint that is not newer than the cutoff commit. A checkpoint at the cutoff commit is ideal, but an older one will do.
2. We need to preserve that checkpoint, and all commits after it, because we need them to create a snapshot as-of the cutoff commit.